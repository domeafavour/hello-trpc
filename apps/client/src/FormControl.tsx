export type FormControlProps = {
  label: string;
  children: React.ReactNode;
};

export function FormControl({ label, children }: FormControlProps) {
  return (
    <div className="flex flex-row gap-2 items-center font-mono">
      {label}:{children}
    </div>
  );
}
