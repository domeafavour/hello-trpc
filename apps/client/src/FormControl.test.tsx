import renderer from 'react-test-renderer';
import { FormControl } from './FormControl';

it('renders <FormControl label="Test Label">Test Child</FormControl> correctly', () => {
  const formControlElement = renderer.create(
    <FormControl label="Test Label">Test Child</FormControl>
  );
  const tree = formControlElement.toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders <FormControl label="Label" /> correctly', () => {
  const formControlElement = renderer.create(
    <FormControl label="Label">hello</FormControl>
  );
  const tree = formControlElement.toJSON();
  expect(tree).toMatchSnapshot();
});
