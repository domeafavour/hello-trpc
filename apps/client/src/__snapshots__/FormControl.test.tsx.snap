// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <FormControl label="Label" /> correctly 1`] = `
<div
  className="flex flex-row gap-2 items-center font-mono"
>
  Label
  :
  hello
</div>
`;

exports[`renders <FormControl label="Test Label">Test Child</FormControl> correctly 1`] = `
<div
  className="flex flex-row gap-2 items-center font-mono"
>
  Test Label
  :
  Test Child
</div>
`;
